includes/sanitizers/class-amp-base-sanitizer.php

```diff
@@ -102,14 +102,18 @@ public function add_or_append_attribute( &$attributes, $key, $value, $separator
 	 * @return string
 	 */
 	public function maybe_enforce_https_src( $src, $force_https = false ) {
+		$https_required = isset( $this->args['require_https_src'] ) && true === $this->args['require_https_src'];
 		$protocol = strtok( $src, ':' );
-		if ( 'https' !== $protocol ) {
+
+		if ( $protocol === $src && ( $https_required || $force_https ) ) {
+			// src has relative protocol, ie //example.com/asdf, so add https.
+			$src = set_url_scheme( $src, 'https' );
```

Comment:
Alright, I took a stab at handling relative paths along with the relative protocol scenario: f7dafe5